lib/src/chart/base/base_chart/initial_animation_mixin.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:fl_chart/src/chart/base/base_chart/initial_animation_configuration.dart';
+import 'package:flutter/material.dart';
+
+@optionalTypeArgs
+mixin InitialAnimationMixin<T, D extends ImplicitlyAnimatedWidget>
+    on AnimatedWidgetBaseState<D> {
+  InitialAnimationConfiguration get initialAnimationConfiguration;
+  Tween<dynamic>? get tween;
+
+  T? _targetValue;
+
+  @override
+  void initState() {
+    super.initState();
+    _handleInitialAnimation();
+  }
+
+  void _handleInitialAnimation() {
+    if (!initialAnimationConfiguration.enabled) {
+      return;
+    }
+
+    if (tween == null || _targetValue == null) {
+      return;
+    }
+    tween!
+      ..begin = tween!.evaluate(animation)
+      ..end = _targetValue;
+
+    controller
+      ..value = 0.0
+      ..forward();
+    didUpdateTweens();
+  }
+
+  T constructInitialData(T data) {
+    if (!initialAnimationConfiguration.enabled) {
+      return data;
+    }
+    _targetValue = data;
+
+    return getAppearanceAnimationData(data);
+  }
+
+  T getAppearanceAnimationData(T data);
+
+  double getAppearanceValue(double minY, double maxY) {
```

Comment:
We have PieChart and RadarChart which are not axis-based charts, so we need to implement an approach that supports all kinds of charts. (In the future, we will add more chart types which are not axis-based. Such as [GaugeChart](https://github.com/imaNNeo/fl_chart/pull/1424))